### ISSUE TITLE
Write line-protocol to defined file if Influx server is not available to avoid data loss

### ISSUE THREAD

**Issue by jensb** [ioBroker/ioBroker.influxdb#112]
I run a Raspi 4b smart home controller with ioBroker and Influx 1.8.6 as storage.
Today I noticed that for some reason influxdb had crashed (bus error), did not successfully restart, and so the system had not collected data for two days. This is very unfortunate since now all average values, predictions and so on are invalid.

**Describe the solution you'd like**
I would like the influx adapter to not just give up if influx is not available, but - optionally - in that case to start writing the data out to a file, ideally in Influx line procotol format, so it can be later easily imported to Influx after the service has been restored. This would avoid data loss in case influx is temporarily not available for storage.
The file can be split every X hours or every X megabytes to prevent it from growing indefinitely. It should automatically be created if it doesn't exist.

**Describe alternatives you've considered**
I have considered using another database, and using flat files in the first place. But all options need a fallback, so this just moves the problem somewhere else.

**Additional context**

**Comment by Apollon77** [comment]
Did you used the buffering? If yes this would have happened already (ok, basically going to mem and not to file), so you should still add a monitoring for your influxdb instance.
Only if buffering is turned off then data will be written directly and if not possible are basically lost. What does the adapter log told?

**Comment by jensb** [comment]
Yes, I actually used buffering, and I actually noticed this just 5 minutes ago by googling some more and finding your post from 2018 here: https://forum.iobroker.net/topic/15235/influxdb-adapter-cache-und-neustart:

    influxdb.0 | 2021-06-02 10:10:07.400 | info | (24746) Store 325833 buffered influxDB history points
    influxdb.0 | 2021-06-02 10:10:07.401 | info | (24746) Too many datapoints (325833) to write at once; write per ID

I did lose some data hoever, by rebooting, and afterwards not enabling the adapter again. So there is some kind of persistent storage in ioBroker that even makes this data survive a reboot, but the adapter needs to stay enabled (well, of course).

This feature is great. Can I buy you a virtual coffee, tee, beer, whatever?

I'm still looking for good monitoring solution that gets me alerted via smartphone when something like this happens. I tried various messenger solutions but none proved to be reliable, texts frequently got lost or delayed.
